src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -433,13 +480,27 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         tabs = _.map(
             this.props.store.genericAssayProfileOptionsByType.result,
             (options, type) => {
+                const molecularProfileIdsInType = options.find(
+                    option =>
+                        option.value ===
+                        this.selectedGenericAssayProfileIdByType.get(type)
+                )!.profileIds;
```

Comment:
i take it we can be sure we will find one because the list was made using ...

Reply:
Yes, because this.selectedGenericAssayProfileIdByType is generated by using `this.props.store.genericAssayProfileOptionsByType.result`, so we can be sure we can always find one.

I will add some comments at here too.